libraries/Microsoft.Bot.Builder.Azure/Storage/BlobsTranscriptStore.cs

```diff
@@ -0,0 +1,419 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Azure.Storage
+{
+    /// <summary>
+    /// The blobs transcript store stores transcripts in an Azure Blob container.
+    /// </summary>
+    /// <remarks>
+    /// Each activity is stored as json blob in structure of
+    /// container/{channelId]/{conversationId}/{Timestamp.ticks}-{activity.id}.json.
+    /// </remarks>
+    public class BlobsTranscriptStore : ITranscriptStore
+    {
+        // Containers checked for creation.
+        private static HashSet<string> _checkedContainers = new HashSet<string>();
+
+        // If a JsonSerializer is not provided during construction, this will be the default static JsonSerializer.
+        private readonly JsonSerializer _jsonSerializer;
+
+        private Lazy<BlobContainerClient> _containerClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlobsTranscriptStore"/> class.
+        /// </summary>
+        /// <param name="dataConnectionString">Azure Storage connection string.</param>
+        /// <param name="containerName">Name of the Blob container where entities will be stored.</param>
+        /// <param name="jsonSerializer">If passing in a custom JsonSerializer, we recommend the following settings:
+        /// <para>jsonSerializer.TypeNameHandling = TypeNameHandling.All.</para>
+        /// <para>jsonSerializer.NullValueHandling = NullValueHandling.Include.</para>
+        /// <para>jsonSerializer.ContractResolver = new DefaultContractResolver().</para>
+        /// </param>
+        public BlobsTranscriptStore(string dataConnectionString, string containerName, JsonSerializer jsonSerializer = null)
+        {
+            if (string.IsNullOrEmpty(dataConnectionString))
+            {
+                throw new ArgumentNullException(nameof(dataConnectionString));
+            }
+
+            if (string.IsNullOrEmpty(containerName))
+            {
+                throw new ArgumentNullException(nameof(containerName));
+            }
+
+            _jsonSerializer = jsonSerializer ?? JsonSerializer.Create(new JsonSerializerSettings
+                                                {
+                                                    // we use All so that we get typed roundtrip out of storage, 
+                                                    // but we don't use validation because we don't know what types are valid
+                                                    TypeNameHandling = TypeNameHandling.All,
```

Comment:
Sorry for getting back on this one, if I understand correctly Blobs requires type info because is used for state and the dialogs need that (oh well), but transcript store is for transcripts and is not related to dialogs, can we use standard json here instead?

Reply:
Updated to:
```cs
            _jsonSerializer = jsonSerializer ?? JsonSerializer.Create(new JsonSerializerSettings
                                                {            
                                                    NullValueHandling = NullValueHandling.Ignore,
                                                    Formatting = Formatting.Indented,
                                                    TypeNameHandling = TypeNameHandling.None,
                                                });
```